pkg/kv/kvclient/kvcoord/dist_sender_rangefeed.go

```diff
@@ -653,6 +653,7 @@ func (ds *DistSender) singleRangeFeed(
 	// Ensure context is cancelled on all errors, to prevent gRPC stream leaks.
 	ctx, cancelFeed := context.WithCancel(ctx)
 	defer func() {
+		ds.metrics.RangefeedRestartRanges.Inc(1)
```

Comment:
This is going to increment the metric when the rangefeed isn't restarted too, e.g. when cancelled.

Reply:
Added check on ctx.Err.

Reply:
It will also increment the metric when `handleRangefeedError` returns a permanent error. Shouldn't this be incremented in `partialRangeFeed` once we've actually determined to restart the rangefeed?

Reply:
Yes, definitely